ivy/functional/backends/tensorflow/manipulation.py

```diff
@@ -356,16 +356,16 @@ def clip(
         if x_max is not None:
             x_max = tf.cast(x_max, promoted_type)
 
-    if x_min is not None and x_max is not None:
+    if cond:
         if tf.size(x) == 0:
             ret = x
         elif x.dtype == tf.bool:
             ret = tf.clip_by_value(tf.cast(x, tf.float16), x_min, x_max)
             ret = tf.cast(ret, x.dtype)
         else:
-            ret = tf.clip_by_value(x, x_min, x_max)
+            ret = tf.experimental.numpy.clip(x, x_min, x_max)
     else:
```

Comment:
I'm not sure if I follow why we need an else case here since `x_min is not None and x_max is not None and x_min > x_max` case (i.e `not cond`) can be handled by `tf.experimental.numpy.clip` as well, right? Maybe I'm  missing something here..

Also, it looks like the frontend tests are failing because of `cond` with the following error:
```python
E     ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

This is because `x_min` and `x_max` can also be arrays.

Reply:
Oh right, sorry missed the fact that they could be arrays too, I think `.any` should be able to resolve this. `tf.experimental.numpy.clip` also follows `tf.clip_by_value` behaviour when `x_min>x_max`,  which we discussed in the above comments. I think the only difference is numpy namespace and the fact that it can accept `None`. You can verify this from the following code.
```>>> print(tf.clip_by_value(tf.constant([1,10,13]),12,11))
tf.Tensor([12 12 12], shape=(3,), dtype=int32)
>>> print(tf.experimental.numpy.clip(tf.constant([1,10,13]),12,11))
tf.Tensor([12 12 12], shape=(3,), dtype=int32)
```

Reply:
> Oh right, sorry missed the fact that they could be arrays too, I think `.any` should be able to resolve this. `tf.experimental.numpy.clip` also follows `tf.clip_by_value` behaviour when `x_min>x_max`, which we discussed in the above comments. I think the only difference is numpy namespace and the fact that it can accept `None`. You can verify this from the following code.
> 
> ```
> >>> print(tf.clip_by_value(tf.constant([1,10,13]),12,11))
> tf.Tensor([12 12 12], shape=(3,), dtype=int32)
> >>> print(tf.experimental.numpy.clip(tf.constant([1,10,13]),12,11))
> tf.Tensor([12 12 12], shape=(3,), dtype=int32)
> ```

Ah, that's interesting. I expected `tf.experimental.numpy.clip` to behave exactly the same as `np.clip`.